pkgs/applications/misc/input-leap/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, qttools, libei ? null, libportal,
+  mkDerivation, pkg-config, openssl, wrapGAppsHook,
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
+}:
+
+mkDerivation rec {
+  pname = "input-leap";
+  version = "unstable-2023-05-24";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "5e2f37bf9ec17627ae33558d99f90b7608ace422";
+    hash = "sha256-55RqdRu/Hi2OTiLjAFJ6Gdgg9iO5NIIJCsOkUQjR9hk=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    curl qtbase avahiWithLibdnssdCompat
+  ] ++ (with xorg; [
+    libX11 libXext libXtst libXinerama libXrandr libXdmcp libICE libSM
+  ]) ++ lib.optionals (libei != null) [
+    libei libportal
+  ];
+  nativeBuildInputs = [ pkg-config cmake wrapGAppsHook qttools ];
+
+  cmakeFlags = [
+    "-DINPUTLEAP_REVISION=${builtins.substring 0 8 src.rev}"
+  ] ++ lib.optional (libei != null) "-DINPUTLEAP_BUILD_LIBEI=ON";
+
+  dontWrapGApps = true;
+
+  # Arguments to be passed to `makeWrapper`, only used by qt5’s mkDerivation
```

Comment:
```suggestion
```

Redundant.

Reply:
I guess at least the original author (@shymega or @KoviRobi?) found this comment worthwhile.

Reply:
I found it useful. I started the original version of my Input Leap derivation, with @KoviRobi's help later on. I don't have any objections to it being removed, but I do wonder if it's worth leaving it there for future maintainers.

Reply:
What will happen if this comment is not included?

It looks like "I am a novice, I should document every single line or else I will forget what this piece of code does" to me.

Reply:
My apologies, I always forget how to avoid double wrapping, so went to the manual https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped and I just used the example from there. Feel free to remove if you want